Update treemap.md (#1201)
Codecademy/docs: content/java/concepts/treemap/treemap.md

  - 'paths/computer-science'
---

> A `TreeMap` uses the [`Map`](https://www.codecademy.com/resources/docs/java/map) interface, which means it stores items as key-value pairs.

Reviewer: ```suggestion
A **`TreeMap`** uses the [`Map`](https://www.codecademy.com/resources/docs/java/map) interface, which means it stores items as key-value pairs.
```

> Similiar to a [`HashMap`](https://www.codecademy.com/resources/docs/java/hashmap), the `TreeMap` stores key-value pairs. The difference is the means of storage each type of collection uses internally. A `HashMap` stores its keys as hashes in a [`Hash Table`](https://www.codecademy.com/resources/docs/general/hash-table) whereas a `TreeMap` stores its keys in a ['Binary Tree Structure'](https://www.codecademy.com/resources/docs/general/binary-search-tree), which is a type of tree data structure that has at most two children nodes per parent. While the structure of a 'TreeMap' is more orderly, it comes as a tradeoff with speed in comparision to a 'HashMap'. Most `HashMap` operations run in ammortized O(1) time (in [Big-O notation](https://www.codecademy.com/resources/docs/general/big-o-notation)) and are generally more time efficent.  Most `TreeMap` operations run in O(log(n)) time and the `TreeMap` is generally more space efficent due to it's dynamic allocation of memory per value added. Do note that complexity is a crucial factor to determining what structure to implement. If you want faster operations over your data, you would want to aim for a `HashMap`. However, if you're looking to keep your values sorted, it's best to stick to a `TreeMap`.

Reviewer: ```suggestion
Similiar to a [`HashMap`](https://www.codecademy.com/resources/docs/java/hashmap), the `TreeMap` stores key-value pairs. The difference is the means of storage each type of collection uses internally. A `HashMap` stores its keys as hashes in a [`Hash Table`](https://www.codecademy.com/resources/docs/general/hash-table) whereas a `TreeMap` stores its keys in a ['Binary Tree Structure'](https://www.codecademy.com/resources/docs/general/binary-search-tree), which is a type of tree data structure that has at most two children nodes per parent.

While the structure of a 'TreeMap' is more orderly, it comes as a tradeoff with speed in comparison to a 'HashMap'. In [Big-O notation](https://www.codecademy.com/resources/docs/general/big-o-notation), most `HashMap` operations run in amortized O(1) time; they are generally more time-efficient.  Most `TreeMap` operations run in O(log(n)) time and the `TreeMap` is generally more space-efficient due to its dynamic allocation of memory per value added.

> **Note:** Complexity is a crucial factor when determining what structure to implement. If you want faster operations over your data, you would want to aim for a `HashMap`. However, if you're looking to keep your values sorted, it's best to stick to a `TreeMap`.
```

I'm not sure I quite follow this part:

> If you want faster operations over your data, you would want to aim for a `HashMap`. However, if you're looking to keep your values sorted, it's best to stick to a `TreeMap`.

Do we mean to say that a `HashMap` is faster but less efficient while a `TreeMap` is not as fast but more efficient?

Author: Yes, a Hashmap is faster but may not have sorted entries. A Treemap would be slower but has more versatility and more locality depending on how the values are ordered.

## Syntax

```pseudo
import java.util.TreeMap

TreeMap<KeyDataType, ValueDataType> myTreeMap = new TreeMap<KeyDataType, ValueDataType>();
```

> The `TreeMap` class comes from the `java.util` package, therefore, it must be imported in order to be used. The `TreeMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDataType` and `ValueDataType` can either be different or the same.

Reviewer: ```suggestion
Since, the `TreeMap` class comes from the `java.util` package, it must be imported in order to be used. The `TreeMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDataType` and `ValueDataType` can either be different or the same.
```

Author: Ooo, should probably fix that little grammar mistakes! Thanks for the catch!

  - 'paths/computer-science'
---

A **`TreeMap`** uses the [`Map`](https://www.codecademy.com/resources/docs/java/map) interface, which means it stores items as key-value pairs.

Similiar to a [`HashMap`](https://www.codecademy.com/resources/docs/java/hashmap), the `TreeMap` stores key-value pairs. The difference is the means of storage each type of collection uses internally. A `HashMap` stores its keys as hashes in a [`Hash Table`](https://www.codecademy.com/resources/docs/general/hash-table) whereas a `TreeMap` stores its keys in a ['Binary Tree Structure'](https://www.codecademy.com/resources/docs/general/binary-search-tree), which is a type of tree data structure that has at most two children nodes per parent.

While the structure of a 'TreeMap' is more orderly, it comes as a tradeoff with speed in comparison to a 'HashMap'. In [Big-O notation](https://www.codecademy.com/resources/docs/general/big-o-notation), most `HashMap` operations run in amortized O(1) time; they are generally more time-efficient.  Most `TreeMap` operations run in O(log(n)) time and the `TreeMap` is generally more space-efficient due to its dynamic allocation of memory per value added.

> > **Note:** Complexity is a crucial factor when determining what structure to implement. If you want faster operations over your data, you would want to aim for a `HashMap`. However, if you're looking to keep your values sorted, it's best to stick to a `TreeMap`.

Reviewer: Is there any way we can rephrase this note so that "you"/"your" is not included in the text? For Docs entries, we prefer that entries have a more neutral, manual tone that just presents information.

Author: @Dusch4593 Good thought, let me change the you/your

## Syntax

```pseudo
import java.util.TreeMap

TreeMap<KeyDataType, ValueDataType> myTreeMap = new TreeMap<KeyDataType, ValueDataType>();
```

> Since, the `TreeMap` class comes from the `java.util` package, it must be imported in order to be used. The `TreeMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDataType` and `ValueDataType` can either be different or the same.

Reviewer: ```suggestion
Since the `TreeMap` class comes from the `java.util` package, it must be imported in order to be used. The `TreeMap` is initialized with two generic types inside angle brackets `< ... >`. The generic data types for `KeyDataType` and `ValueDataType` can either be different or the same.
```
